src/bpmncwpverify/builder/cwp_builder.py

```diff
@@ -0,0 +1,127 @@
+from typing import List
+from abc import ABC, abstractmethod
+from xml.etree.ElementTree import Element
+from bpmncwpverify.core.state import SymbolTable
+from bpmncwpverify.visitors.cwp_connectivity_visitor import CwpConnectivityVisitor
+from returns.result import Result, Success
+from bpmncwpverify.error import Error
+from bpmncwpverify.core.cwp import Cwp, CwpEdge, CwpState
+from bpmncwpverify.core.expr import ExpressionListener
+from returns.pipeline import is_successful
+from returns.functions import not_
+
+
+class CwpBuilder(ABC):
+    @abstractmethod
+    def build_state(self, element: Element) -> None:
```

Comment:
`set_edge` or `add_edge` would be more inline with a `Builder` interface. So `builder.add_edge` or `builder.set_edge`. The same with `build_state`. Make it `add` or `set` state. Then when all the edges and states are _added_ or _setted_, then `builder.build()` to get the instance of the Cwp. That instance is guaranteed to be connected, have start states, have end states, and be type correct.